InvoiceProcessor.Application/Queries/Outbox/GetPendingProcess/QueryHandler.cs

```diff
@@ -12,19 +12,36 @@ public partial class GetPendingProcess
         public class QueryHandler : IRequestHandler<Query, List<Model>>
         {
             private readonly IOutboxStorage _outboxStorage;
-            private readonly IMapper _mapper;
 
-            public QueryHandler(IOutboxStorage outboxStorage, IMapper mapper)
+            public QueryHandler(IOutboxStorage outboxStorage)
             {
                 _outboxStorage = outboxStorage;
-                _mapper = mapper;
             }
 
             public async Task<List<Model>> Handle(Query request, CancellationToken cancellationToken)
             {
                 var pendingItems = await _outboxStorage.GetPendingItemsAsync(cancellationToken);
-                
-                return _mapper.Map<List<Model>>(pendingItems);
+                var result = PopulateData(pendingItems);
```

Comment:
Insisted of doing the conversation manually, you can use Linq. The code would be more readable. And .net would select the best algorithm, which will help to improve the performance.
Ex: ```return pendingItems.Select(m => new Model(m.Guid.Value, m.CommandType, m.Data)).ToList();```